rmw_fastrtps_cpp/src/rmw_node.cpp

```diff
@@ -166,16 +166,24 @@ get_security_file_paths(
   const char * file_names[3] = {"ca.cert.pem", "cert.pem", "key.pem"};
   size_t num_files = sizeof(file_names) / sizeof(char *);
 
-  const char * file_prefix = "file://";
+  std::string file_prefix("file://");
 
-  std::string tmpstr;
   for (size_t i = 0; i < num_files; i++) {
-    tmpstr = std::string(rcutils_join_path(node_secure_root, file_names[i]));
-    if (!rcutils_is_readable(tmpstr.c_str())) {
+    const char * file_path = rcutils_join_path(node_secure_root, file_names[i]);
```

Comment:
Really, `rcutils_join_path` should not return `const char *` but just `char *` instead of needing to const cast it to free the result...

Reply:
Yeah, I agree with that.  I'll do a separate PR with that, since every place that uses it has to do the same thing.

Reply:
ros2/ros2#423